magma/passes/insert_coreir_wires.py

```diff
@@ -0,0 +1,89 @@
+from .passes import DefinitionPass
+from ..digital import Digital
+from ..generator import Generator
+from ..circuit import Circuit, DeclareCoreirCircuit
+from ..t import In, Out
+from ..array import Array
+from ..tuple import Tuple
+from ..bits import Bits
+
+
+def simulate_wire(self, value_store, state_store):
+    value_store.set_value(self.O, value_store.get_value(self.I))
+
+
+class Wire(Generator):
+    @staticmethod
+    def generate(name, T):
+        if issubclass(T, Digital):
```

Comment:
probably can be a ternary: `width = 1 if issubclass(T, Digital) else T.flat_length()`